src/lwc/rd2EntryForm/rd2EntryForm.js

```diff
@@ -0,0 +1,318 @@
+import FIELD_INSTALLMENT_FREQUENCY from '@salesforce/schema/npe03__Recurring_Donation__c.InstallmentFrequency__c';
+import FIELD_DAY_OF_MONTH from '@salesforce/schema/npe03__Recurring_Donation__c.Day_of_Month__c';
+import FIELD_START_DATE from '@salesforce/schema/npe03__Recurring_Donation__c.StartDate__c';
+import FIELD_CAMPAIGN from '@salesforce/schema/npe03__Recurring_Donation__c.npe03__Recurring_Donation_Campaign__c';
+
+import cancelButtonLabel from '@salesforce/label/c.stgBtnCancel';
+import saveButtonLabel from '@salesforce/label/c.stgBtnSave';
+import unknownErrorLabel from '@salesforce/label/c.commonUnknownError';
+import newHeaderLabel from '@salesforce/label/c.RD2_EntryFormHeader';
+import editHeaderLabel from '@salesforce/label/c.commonEdit';
+import currencyFieldLabel from '@salesforce/label/c.lblCurrency';
+import donationSectionHeader from '@salesforce/label/c.RD2_EntryFormDonationSectionHeader';
+import donorSectionHeader from '@salesforce/label/c.RD2_EntryFormDonorSectionHeader';
+import otherSectionHeader from '@salesforce/label/c.RD2_EntryFormOtherSectionHeader';
+import insertSuccessMessage from '@salesforce/label/c.RD2_EntryFormInsertSuccessMessage';
+import updateSuccessMessage from '@salesforce/label/c.RD2_EntryFormUpdateSuccessMessage';
+
+import getSetting from '@salesforce/apex/RD2_entryFormController.getSetting';
+
+export default class rd2EntryForm extends LightningElement {
+
+    @api parentId;
+    @api recordId;
+    listenerEvent = 'rd2EntryFormEvent';
+
+    customLabels = Object.freeze({
+        cancelButtonLabel,
+        saveButtonLabel,
+        donationSectionHeader,
+        donorSectionHeader,
+        otherSectionHeader
+    });
+
+    @track fields = {};
+    @track record;
+    recurringDonationInfo;
+    fieldInfos;
+
+    @track header = newHeaderLabel;
+
+    @track isAutoNamingEnabled;
+    @track isMultiCurrencyEnabled;
+
+    @track contactId;
+    @track accountId;
+
+    @track isLoading = true;
+    @track isRecordReady = false;
+    isSettingReady = false;
+
+    @track hasError = false;
+    @track errorMessage = {};
+
+    /*******************************************************************************
+    * @description Dynamic render edit form CSS to show/hide the edit form 
+    */
+    get cssEditForm() {
+        return (!this.isLoading && this.isSettingReady && this.isRecordReady)
+            ? ''
+            : 'slds-hide';
+    }
+
+    /*******************************************************************************
+    * @description Get Recurring Donation SObject API name 
+    */
+    get recurringDonationName() {
+        return getNestedProperty(RECURRING_DONATION_INFO, 'objectApiName');
+    }
+
+
+    /*******************************************************************************
+    * @description Retrieve Recurring Donation Object info
+    */
+    @wire(getObjectInfo, {objectApiName: '$recurringDonationName' })
```

Comment:
Instead of using the `recurringDonationName` can this be written as 
` @wire(getObjectInfo, { objectApiName: RECURRING_DONATION_OBJECT.objectApiName })`? If so, we can remove functions for the `recurringDonationName`.